templates/deployment.yaml

```diff
@@ -0,0 +1,80 @@
+apiVersion: apps/v1
+kind: Deployment                
+metadata:
+  name: {{ include "openvpn-as.fullname" . }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  strategy:
+    type: Recreate
+  selector:
+    matchLabels:
+      app: {{ include "openvpn-as.name" . }}
+  template:
+    metadata:
+      labels:
+        app: {{ include "openvpn-as.name" . }}
+    spec:
+      securityContext:
+        runAsUser: 0
+      containers:
+        - name: {{ .Chart.Name }}
+          image: {{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - name: admin-gui
+              containerPort: {{ .Values.service.ports.admin }}
+              protocol: TCP
+            - name: vpn-tcp
+              containerPort: {{ .Values.service.ports.tcp }}
+              protocol: TCP
+            - name: vpn-udp
+              containerPort: {{ .Values.service.ports.udp }}
+              protocol: UDP
+          securityContext:
+            privileged: {{ .Values.securityContext.privileged }}
+            capabilities:
+              add: {{ toJson .Values.securityContext.capabilities.add }}
+          resources:
+            {{- toYaml .Values.resources | nindent 12 }}
+          {{- if and .Values.postStart.enabled }}
+          lifecycle:
+            postStart:
+              exec:
+                command: ["/bin/sh", "-c", "/scripts/configure.sh 2>&1 >> /var/log/k8s-post-start.log"]
+          {{- end }}
+          livenessProbe:
+            initialDelaySeconds: 100
+            timeoutSeconds: 5
+            exec:
+              command: ["/usr/local/openvpn_as/scripts/sacli", "Status"]
+          readinessProbe:
+            initialDelaySeconds: 100
+            tcpSocket:
+              port: {{ .Values.service.ports.admin }}
+          volumeMounts:
+            - name: openvpn-data
+              mountPath: /openvpn
+            - name: dev-net-tun
+              mountPath: /dev/net/tun
+            {{- if .Values.postStart.enabled }}
+            - name: poststart-script
+              mountPath: /scripts
+              readOnly: true
+            {{- end }}
+      volumes:
+        - name: openvpn-data
+          persistentVolumeClaim:
+            claimName: {{ include "openvpn-as.fullname" . }}-pvc
+        - name: dev-net-tun
+          hostPath:
+            path: /dev/net/tun
+        {{- if and .Values.postStart.enabled }}
+        - name: poststart-script
+          secret:
+            secretName: {{- if .Values.postStart.secretRefName }}
+                          {{ .Values.postStart.secretRefName }}
+                        {{- else }}
+                          {{ include "openvpn-as.fullname" . }}-poststart
+                        {{- end }}
+            defaultMode: 0755
```

Comment:
Not sure whether that makes a practical difference but wouldn't `700` be more appropriate for a file that comes from a secret?